packages/react-composites/src/composites/ChatComposite/file-sharing/FileCard.tsx

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { Icon, mergeStyles, ProgressIndicator, Stack, Text, useTheme } from '@fluentui/react';
+import { getFileTypeIconProps } from '@fluentui/react-file-type-icons';
+import React from 'react';
+
+/**
+ * @beta
+ */
+export interface FileCardProps {
+  /**
+   * File name.
+   */
+  fileName: string;
+  /**
+   * Extension of the file used for rendering the file icon.
+   */
+  fileExtension: string;
+  /**
+   * File upload/download progress percentage between 0 and 1.
+   * Not shown if undefined or 1.
+   */
+  progress?: number;
+  /**
+   * Icon to display for actions like download, upload, etc. on the right of file name.
+   */
+  actionIcon?: JSX.Element;
+  /**
+   * Function that runs when actionIcon is clicked
+   */
+  actionHandler?: () => void;
+}
+
+/**
+ * @beta
+ */
+export const FileCard = (props: FileCardProps): JSX.Element => {
+  const { fileName, fileExtension, progress, actionIcon } = props;
+  const theme = useTheme();
+
+  return (
+    <Stack
+      className={mergeStyles({
+        width: '14rem',
```

Comment:
Avoid inline constants like this. Please define a styles constant that is merged in.

Reply:
I want to understand why we want to avoid it?
Is there a strong reason?

Reply:
Same as discussion above with @JamesBurnside already. Let's just discuss it there.

First off, constants help discoverability a little bit (magic constants in the code are extremely hard to find. What if we wanted to change all `rem` to `em` (not that we should want such a thing...)). Second, it's a good practice because it allows pure components to skip a render if the other props are unchanged as well.